chain/cosmos/chain_node.go

```diff
@@ -961,11 +961,25 @@ func (tn *ChainNode) InstantiateContract(ctx context.Context, keyName string, co
 }
 
 // ExecuteContract executes a contract transaction with a message using it's address.
-func (tn *ChainNode) ExecuteContract(ctx context.Context, keyName string, contractAddress string, message string, extraExecTxArgs ...string) (txHash string, err error) {
+func (tn *ChainNode) ExecuteContract(ctx context.Context, keyName string, contractAddress string, message string, extraExecTxArgs ...string) (res *types.TxResponse, err error) {
 	cmd := []string{"wasm", "execute", contractAddress, message}
 	cmd = append(cmd, extraExecTxArgs...)
 
-	return tn.ExecTx(ctx, keyName, cmd...)
+	txHash, err := tn.ExecTx(ctx, keyName, cmd...)
+	if err != nil {
+		return &types.TxResponse{}, err
+	}
+
+	txResp, err := tn.GetTransaction(tn.CliContext(), txHash)
+	if err != nil {
+		return &types.TxResponse{}, fmt.Errorf("failed to get transaction %s: %w", txHash, err)
+	}
+
+	if txResp.Code != 0 {
+		return txResp, fmt.Errorf("error in transaction (code: %d): %s", txResp.Code, txResp.RawLog)
```

Comment:
@srdtrk is this what you were looking for?

Reply:
I need a v7 backport of this to try out with my test case. It'd be nice if you could provide me with the go.mod import i'd need to use

Reply:
I have not tested it on v7 yet, but here is the cherry picked commit:

99ec15f2674d326aa0ab84138e189141c31b0204 - `github.com/strangelove-ventures/interchaintest/v7 v7.0.0-20231025163603-99ec15f2674d`

Reply:
Tested it and it works!